benchmark/scripts/compare_perf_tests.py

```diff
@@ -18,40 +18,270 @@
+        self.max_rss = int(csv_row[8])  # Maximum Resident Set Size (B)
+        # TODO if we really want to compute mean MAX_RSS: self.S_memory
+
+    @property
+    def sd(self):  # Standard Deviation (ms)
+        return (0 if self.samples < 2 else
+                sqrt(self.S_runtime / (self.samples - 1)))
+
+    @property
+    def cv(self):  # Coeficient of Variation (%)
+        return self.sd / self.mean
+
+    # Compute running variance, B. P. Welford's method
+    # See Knuth TAOCP vol 2, 3rd edition, page 232, or
+    # https://www.johndcook.com/blog/standard_deviation/
+    # M is mean, Standard Deviation is defined as sqrt(S/k-1)
+    @staticmethod
+    def running_mean_variance((k, M_, S_), x):
+        k = float(k + 1)
+        M = M_ + (x - M_) / k
+        S = S_ + (x - M_) * (x - M)
+        return (k, M, S)
+
+    def merge(self, r):
+        self.min = min(self.min, r.min)
+        self.max = max(self.max, r.max)
+        # self.median = None # unclear what to do here
+
+        def push(x):
+            state = (self.samples, self.mean, self.S_runtime)
+            state = self.running_mean_variance(state, x)
+            (self.samples, self.mean, self.S_runtime) = state
+
+        # Merging test results with up to 3 samples is exact
+        # TODO investigate how to best handle merge of higher sample counts
+        values = [r.min, r.max, r.median, r.mean][:min(r.samples, 4)]
+        map(push, values)
+
+    header = ('TEST', 'MIN', 'MAX', 'MEAN', 'MAX_RSS')
+
+    # Tuple of values formatted for display in results table:
+    # (name, min value, max value, mean value, max_rss)
+    def values(self):
+        return (self.name, str(self.min), str(self.max),
+                str(self.mean), str(self.max_rss))
+
+
+class ResultComparison:
+    def __init__(self, old, new):
+        self.old = old
+        self.new = new
+        self.name = old.name  # Test name, convenience accessor
+
+        # Speedup ratio
+        self.ratio = (old.min + 0.001) / (new.min + 0.001)
+
+        # Test runtime improvement in %
+        ratio = (new.min + 0.001) / (old.min + 0.001)
+        self.delta = ((ratio - 1) * 100)
+
+        self.is_dubious = (  # FIXME this is legacy
+            ' (?)' if ((old.min < new.min and new.min < old.max) or
+                       (new.min < old.min and old.min < new.max))
+            else '')
+
+    header = ('TEST', 'OLD', 'NEW', 'DELTA', 'SPEEDUP')
+
+    # Tuple of values formatted for display in results table:
+    # (name, old value, new value, delta [%], speedup ratio)
+    def values(self):
+        return (self.name, str(self.old.min), str(self.new.min),
+                '{0:+.1f}%'.format(self.delta),
+                '{0:.2f}x{1}'.format(self.ratio, self.is_dubious))
+
+
+class TestComparator:
+    def __init__(self, old_file, new_file, delta_threshold, changes_only):
+
+        def load_from_CSV(filename):
+            def skip_totals(row):
+                        return len(row) > 8
```

Comment:
Benchmark output contains exactly 8 columns, so all of them get skipped because of this line, I believe.